addons/survey/models/survey_user.py

```diff
@@ -265,12 +283,44 @@ def _get_line_answer_values(self, question, answer, answer_type):
 
         if answer_type == 'suggestion':
             vals['suggested_answer_id'] = int(answer)
+            answer_score = self._get_speed_rating_score(answer)
+            if answer_score:
+                vals.update({'answer_score': answer_score})
```

Comment:
Currently answer_score is computed at create / write. I would keep this logic, aka low-level computation in those CRUD methods (may become a computed field but probably simpler to do the computation manually). That way we have a single entry point for score computation. _get_speed_rating_score could be merged in a _compute_answer_score or _get_answer_score (to avoid messing with compute keyword).